### PR TITLE
Add spark check_overflow function and cast toDecimal32/64/128

### PR DESCRIPTION
**Pull request by loneylee**
Changelog category (leave one):
- New Feature
- Improvement
- Bug Fix (user-visible misbehaviour in official stable or prestable release)
- Performance Improvement
- Backward Incompatible Change
- Build/Testing/Packaging Improvement
- Documentation (changelog entry is not required)
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by kyligence-git** [comment]
Can one of the admins verify this patch?

**Comment by loneylee** [comment]
retest this please

**Comment by loneylee** [comment]
test this please with 691

File: utils/local-engine/Parser/SerializedPlanParser.cpp [hunk 1]
```diff
@@ -182,6 +182,35 @@ std::shared_ptr<DB::ActionsDAG> SerializedPlanParser::expressionsToActionsDAG(
     return actions_dag;
 }
 
+std::string getDecimalFunction(const substrait::Type_Decimal & decimal, const bool nullOnOverflow) {
```
Comment (liuneng1994) [inline utils/local-engine/Parser/SerializedPlanParser.cpp:185]: 参数名小写

Comment (loneylee) [reply utils/local-engine/Parser/SerializedPlanParser.cpp]: fix

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Pull request merged**